packages/pds/src/sql-blockstore.ts

```diff
@@ -21,36 +27,58 @@ export class SqlBlockstore extends IpldStore {
   }
 
   async getBytes(cid: CID): Promise<Uint8Array> {
+    const foundStaged = this.staged.get(cid.toString())
+    if (foundStaged) return foundStaged
     const found = await this.db.db
       .selectFrom('ipld_block')
       .where('cid', '=', cid.toString())
       .select('content')
       .executeTakeFirst()
-    if (!found) throw new Error(`Not found: ${cid.toString()}`)
-    return found.content
+    if (found) return found.content
+    throw new Error(`Not found: ${cid.toString()}`)
+  }
+
+  async stageBytes(k: CID, v: Uint8Array): Promise<void> {
+    this.staged.set(k.toString(), v)
   }
 
-  async putBytes(cid: CID, bytes: Uint8Array): Promise<void> {
+  async saveStaged(): Promise<void> {
     this.db.assertTransaction()
-    const insertBlock = this.db.db
-      .insertInto('ipld_block')
-      .values({
+    const blocks: IpldBlock[] = []
+    const creators: IpldBlockCreator[] = []
+    // const promises: Promise<any>[] = []
+    for (const staged of this.staged.entries()) {
+      const [cid, bytes] = staged
+      blocks.push({
         cid: cid.toString(),
         size: bytes.length,
         content: bytes,
         indexedAt: this.timestamp || new Date().toISOString(),
       })
+      creators.push({
+        cid: cid.toString(),
+        did: this.did,
+      })
+    }
+    const insertBlocks = this.db.db
+      .insertInto('ipld_block')
+      .values(blocks)
       .onConflict((oc) => oc.doNothing())
       .execute()
-    const insertBlockOwner = this.db.db
+    const insertCreators = this.db.db
```

Comment:
Feels good to see creators tracked only for inserts to the blockstore that are likely to live under a real commit for them.